internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -443,13 +450,27 @@ func (b *BeatsMonitor) injectLogsInput(cfg map[string]interface{}, components []
 								"to":   "data_stream.dataset",
 							},
 						},
+						"when": map[string]any{
```

Comment:
isn't the above `drop_fields` making this when condition redundant?

Reply:
Done on 7f3b86f3db